gdal/doc/source/development/rfc/rfc77_drop_python2_support.rst

```diff
@@ -0,0 +1,229 @@
+======================= =============== ===============  ===============================
+
+https://wiki.python.org/moin/Python3LinuxDistroPortingStatus
+
+
+Windows:
++++++++++
+
+* Conda: Python 3.6-3.9
+* OSGeo4W: Python 3.7
+* gisinternals: Python 2.7, 3.4-3.7
+
+MacOS:
++++++++
+
+It appears the even though Python 2.7 is preinstalled on Mac OS X,
+Python 3.5-3.9 can be installed alongside Python 2 on Mac OS X 10.8 (Released: July 2012) and newer.
+
+https://docs.python.org/3.6/using/mac.html.
+
+Which version should be the new minimum version ?
+-----------------------------------------------------
+
+This RFC suggest the new minimum supported Python version should be 3.6 for the following reasons:
+
+* Python < 3.6 is End of Life.
+* Long list of great new features that were introduced in Python 3.6, Several of which are immediately useful to simplify code or improve testing:
+    * f-strings
+    * builtin pathlib
+    * underscores in numeric literals
+    * type annotations
+    * malloc debugging
+* Python 3.6 is supported out of the box in virtually every relevant OS.
+* Python 3.6 is probably the safest choice for now in respect to other related projects.
+* Python 3.7 (and newer) isn't available seamlessly in some popular LTS Linux distributions.
+* We want to make the transition as smooth and easy as possible. Setting the minimum to Python 3.7 might make the transition harder for the CI because of the above reason.
+* Dropping Python 3.6 in favour of Python 3.7 or newer in future versions shouldn't be as hard as this drop (see next section for a suggested approach).
+
+GDAL Release cycle and regular Python version dropping
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+When releasing GDAL 3.1.0, Even Rouault suggested GDAL would use fixed release cycles of 6 months between major versions:
+
+http://osgeo-org.1560.x6.nabble.com/gdal-dev-Reconsidering-release-cycle-length-td5436163.html#a5436242
+
+Projecting from that suggestion, GDAL 3.3.0 should be released around April-May 2021.
+
+We could potentially synchronize with NEP 29 -
+Recommend Python and Numpy version support as a community policy standard.
+Which suggests when to drop each Python version.
+We could consider a more conservative approach and delay each drop by a few more months.
+
+https://numpy.org/neps/nep-0029-deprecation_policy.html
+
+NEP 29 suggests to drop support for Python 3.6 support on Jun 23, 2020 (in favour of Python 3.7).
+Further discussion on this matter is a subject for another RFC.
+
+Backward compatibility
+----------------------
+
+Currently, GDAL Python code itself is compatible with Python 2 and Python 3.
+Once this PR is accepted, GDAL 3.3.0 would not be compatible with Python 2.
+Thus any "Python 2 only" code that uses GDAL would need to be upgraded to Python 3 and
+at the same time the respective Python interpreter would need to be upgraded
+to a supported Python version.
+
+Making GDAL 3.2.0 as a LTS version:
```

Comment:
should be rephrased. On a quick read, one might understand that 3.2.0 will be a LTS version

Reply:
right. done.